base/promotion.jl

```diff
@@ -23,11 +23,16 @@ typejoin(@nospecialize(t), @nospecialize(s), @nospecialize(u)) = (@_foldable_met
 typejoin(@nospecialize(t), @nospecialize(s), @nospecialize(u), ts...) = (@_foldable_meta; @_nospecializeinfer_meta; afoldl(typejoin, typejoin(t, s, u), ts...))
 function typejoin(@nospecialize(a), @nospecialize(b))
```

Comment:
why not make this `function typejoin(@nospecialize(a::Union{Type, TypeVar}), @nospecialize(b::Union{Type, TypeVar}))` making `typejoin(1,"hi")` return `Any` just feels weird...